### ISSUE TITLE
when I use Julia to call tvm, the error : CommandLine Error: Option 'aarch64-enable-ccmp' registered more than once! LLVM ERROR: inconsistency in registered CommandLine options

### ISSUE THREAD

**Issue by agdkyang** [JuliaLang/julia#36342]
when I use Julia to call tvm:
the error : CommandLine Error: Option 'aarch64-enable-ccmp' registered more than once! LLVM ERROR: inconsistency in registered CommandLine options

**Comment by agdkyang** [comment]
I am not sure if the reason is that the version of llvm for julia and tvm is  different.

**Comment by StefanKarpinski** [comment]
Julia does require it's own fairly specific build of LLVM. I don't know what LLVM tvm requires, but that could be an issue. It would be helpful if you could give a minimal (non-)working example. Otherwise, I'm afraid there's not much actionable here.

**Comment by vchuravy** [comment]
You might want to try building TVM against Julia's LLVM using BinaryBuilder (see https://github.com/JuliaPackaging/Yggdrasil/tree/master/L/LLVM for the LLVM build scripts we use). In general we are using symbol versioning to reduce potential conflicts, but I have seen cases where weak symbols ignore the symbol versions and then link against the wrong library,

**Comment by agdkyang** [comment]
> Julia does require it's own fairly specific build of LLVM. I don't know what LLVM tvm requires, but that could be an issue. It would be helpful if you could give a minimal (non-)working example. Otherwise, I'm afraid there's not much actionable here.

Yeah, I am not sure if the problem is caused by  the different llvm version for Julia and tvm.

**Comment by agdkyang** [comment]
> You might want to try building TVM against Julia's LLVM using BinaryBuilder (see https://github.com/JuliaPackaging/Yggdrasil/tree/master/L/LLVM for the LLVM build scripts we use). In general we are using symbol versioning to reduce potential conflicts, but I have seen cases where weak symbols ignore the symbol versions and then link against the wrong library,

@vchuravy I want LLVM_full@8.0.0, but I can't find it in your link. Could you please tell me where do you find the LLVM patch in the website?

**Comment by jpsamaroo** [comment]
@vchuravy does the solution to this require changes to libc, or something else?

**Comment by vchuravy** [comment]
Depends on what the actual issue is.. The question to ask is: Why are the symbol suffixes ignored. The last time I delved into this issue it turned out that weak symbols in libclang ignore the suffixes. The fact that LLVM fails during command line argument parsing hints at a second issue. LLVM uses a static variable here and it seems like those are shared between differing LLVM builds. So the bigger question is: How do we effectively namespace two dynamic libraries and convince the dynamic linker that ourLLVM really is different from the system LLVM.

One avenue I have thought about is to link LLVM statically to Julia, the only caveat being that we would need to re-export the LLVM C-API so that LLVM.jl could still do it's thing.
